imap/conversations.h

```diff
@@ -269,6 +269,19 @@ extern int conversations_iterate_searchset(struct conversations_state *state,
 extern conversation_id_t conversations_guid_cid_lookup(struct conversations_state *state,
                                                        const char *guidrep);
 
+/* lookup the matching name or uniqueid */
+#define conv_guidrec_mboxname(rec) conversations_folder_mboxname((rec)->cstate, (rec)->foldernum)
+#define conv_guidrec_uniqueid(rec) conversations_folder_uniqueid((rec)->cstate, (rec)->foldernum)
+#define conv_guidrec_mbentry(rec, mbentryp) ( \
+  ((rec)->version > CONV_GUIDREC_BYNAME_VERSION) ? \
+   mboxlist_lookup_by_uniqueid(conv_guidrec_uniqueid(rec), mbentryp, NULL) : \
+   mboxlist_lookup(conv_guidrec_mboxname(rec), mbentryp, NULL) \
+ )
+#define conv_guidrec_mboxcmp(rec, mbox) ( \
+  ((rec)->version > CONV_GUIDREC_BYNAME_VERSION) ? \
```

Comment:
these makes the macros arguments potentially be evaluated twice

Reply:
yeah, all these similar things do.  We never call them with an expression, and I'm not sure how to make them safe against that while still returning a single value...

https://stackoverflow.com/questions/2679182/have-macro-return-a-value

Without using either an inline function or a GCC extension...